src/chainstate/stacks/boot/pox-2.clar

```diff
@@ -761,6 +758,65 @@
     })
 )
 
+(define-private (increase-reward-cycle-entry 
+                  (reward-cycle-index uint)
+                  (updates (optional { first-cycle: uint, reward-cycle: uint, stacker: principal, add-amount: uint })))
+    (let ((data (try! updates))
+          (first-cycle (get first-cycle data))
+          (reward-cycle (get reward-cycle data)))
+    (if (> first-cycle reward-cycle)
+        ;; not at first cycle to process yet
+        (some { first-cycle: first-cycle, reward-cycle: (+ u1 reward-cycle), stacker: (get stacker data), add-amount: (get add-amount data) })
+        (let ((existing-entry (unwrap-panic (map-get? reward-cycle-pox-address-list { reward-cycle: reward-cycle, index: reward-cycle-index })))
+              (existing-total (unwrap-panic (map-get? reward-cycle-total-stacked { reward-cycle: reward-cycle }))))
+            ;; stacker must match
+            (asserts! (is-eq (get stacker existing-entry) (some (get stacker data))) none)
```

Comment:
If `(is-some (get stacker data))` is true, then wouldn't this assertion's failure be a Really Bad Thing, i.e. indicative that `reward-cycle-pox-address-list` was corrupted somehow, since the `reward-cycle-index` didn't refer to a PoX address from this stacker?

I can see why `(get stacker data)` could be `none` -- e.g. when the stacker had stacked through a delegate.  But, should this function even be called when this is the case (or, can we not tell prior to calling that this is the case)?

Reply:
Yes, this would be a Really Bad Thing, but Clarity cannot assert a node panic, so instead, we just panic the transaction.